### ISSUE TITLE
Missing colour to the end in bold lines in help

### ISSUE THREAD

**Issue by jo-so** [neomutt/neomutt#4203]
## Expected Behaviour

Lines should be coloured to the end.

## Actual Behaviour

![image](https://github.com/neomutt/neomutt/assets/23032146/bf6acad3-caa1-4ab2-8cf3-d695fb58f72d)

## Steps to Reproduce

Run `neomutt -F /dev/null -e 'color normal white blue' -e 'push ?'`

## How often does this happen?

- Always

## When did it start to happen?

I can't remember

## NeoMutt Version

```
% neomutt -v
NeoMutt 20231103
Copyright (C) 1996-2022 Michael R. Elkins und andere.
NeoMutt übernimmt KEINERLEI GEWÄHRLEISTUNG. Starten Sie 'neomutt -vv', um
weitere Details darüber zu erfahren. NeoMutt ist freie Software.
Sie können es unter bestimmten Bedingungen weitergeben; starten Sie
'neomutt -vv' für weitere Details.

System: Linux 6.7.9-amd64 (x86_64)
ncurses: ncurses 6.4.20240113 (compiled with 6.4.20240113)
libidn2: 2.3.7 (compiled with 2.3.7)
GPGME: 1.18.0
GnuTLS: 3.8.3
libnotmuch: 5.6.0
PCRE2: 10.42 2022-12-11
storage: tokyocabinet, lmdb
compression: lz4, zlib, zstd

Configure options: --build=x86_64-linux-gnu --prefix=/usr {--includedir=${prefix}/include} {--mandir=${prefix}/share/man} {--infodir=${prefix}/share/info} --sysconfdir=/etc --localstatedir=/var --disable-option-checking --disable-silent-rules {--libdir=${prefix}/lib/x86_64-linux-gnu} --runstatedir=/run --disable-maintainer-mode --disable-dependency-tracking --mandir=/usr/share/man --libexecdir=/usr/libexec --with-mailpath=/var/mail --autocrypt --full-doc --gnutls --gpgme --gsasl --gss --lmdb --lua --lz4 --notmuch --mixmaster --pcre2 --sqlite --tokyocabinet --zlib --zstd

Compilation CFLAGS: -g -O2 -ffile-prefix-map=/build/reproducible-path/neomutt-20231103+dfsg1=. -fstack-protector-strong -fstack-clash-protection -Wformat -Werror=format-security -fcf-protection -fno-delete-null-pointer-checks -D_ALL_SOURCE=1 -D_GNU_SOURCE=1 -D__EXTENSIONS__ -D_XOPEN_SOURCE_EXTENDED -I/usr/include/lua5.4 -I/usr/include -I/usr/include -I/usr/include -DNCURSES_WIDECHAR -I/usr/include -I/usr/include/p11-kit-1 -I/usr/include -isystem /usr/include/mit-krb5 -O2

Standard Einstellungen:
  +attach_headers_color +compose_to_sender +compress +cond_date +debug 
  +encrypt_to_self +forgotten_attachments +forwref +ifdef +imap +index_color 
  +initials +limit_current_thread +multiple_fcc +nested_if +new_mail +nntp +pop 
  +progress +quasi_delete +regcomp +reply_with_xorig +sensible_browser +sidebar 
  +skip_quoted +smtp +status_color +timeout +tls_sni +trash 

Einstellungen bei der Kompilierung:
  +autocrypt +fcntl -flock -fmemopen +futimens +getaddrinfo +gnutls +gpgme 
  +gsasl +gss +hcache -homespool +idn +inotify -locales_hack +lua +mixmaster 
  +nls +notmuch -openssl +pcre2 +pgp -sasl +smime +sqlite +sun_attachment 
  +truecolor 

MAILPATH="/var/mail"
MIXMASTER="mixmaster"
PKGDATADIR="/usr/share/neomutt"
SENDMAIL="/usr/sbin/sendmail"
SYSCONFDIR="/etc"

Um mehr über NeoMutt zu erfahren, besuchen Sie: https://neomutt.org
Falls Sie einen Bug in NeoMutt finden, eröffnen Sie bitte ein Ticket unter:
    https://github.com/neomutt/neomutt/issues
oder senden Sie eine E-Mail an: <neomutt-devel@neomutt.org>
```

## Extra Info

* Operating System and its version **Debian unstable**
* Were you using multiple copies of NeoMutt at once? **No**
* Were you using 'screen' or 'tmux'? **No**
* Is your email local (maildir) or remote (IMAP)? **both**

**Comment by flatcap** [comment]
Thanks for the detailed bug report, @jo-so.

Please can you give branch [[devel/help]](https://github.com/neomutt/neomutt/tree/devel/help) a try (PR: #4204)

Thanks

**Comment by jo-so** [comment]
Yes, this commit fixes this issue.

But I found at lease one more. The background of the prompt `Recall postponed message? ([yes]/no/?):` is black and not blue. Should I open a new issue?

**Comment by flatcap** [comment]
Great...

> The background of the prompt `Recall postponed message? ([yes]/no/?):` is black and not blue

Ah

> Should I open a new issue?

No need.  I'll have a look in the morning.
Thanks

**Comment by flatcap** [comment]
Or I may do it now :-)
I've pushed another commit to the same branch that should fix it.

**Comment by jo-so** [comment]
Yes, that looks fine.

I've looked around and found more places:

* `PGP (e)ncrypt, (s)ign, sign (a)s, (b)oth, (i)nline format, or (c)lear?`
* I'm unsure about messages like `Mail not sent` and `New mail in`
* The message list `M` looks like this 
![image](https://github.com/neomutt/neomutt/assets/23032146/eeb69f4d-7309-4d12-9dde-10d21cf84023)

**Comment by flatcap** [comment]
OK, 6282135c0 should fix the multi-choice prompts.

The "messages" look ok to me.  If I,
```
color message yellow default
```

then hit <kbd>j</kbd> (`<next-entry>`) at the bottom of the Index, I see "**You are on the last message**" in yellow.

If I hit <kbd>M</kbd> (`<show-log-messages>`), I see the same message in the same colour.

If your messages have a black background, you may want to check that your config is using `default` as the background.

**Comment by flatcap** [comment]
[Background]
Not long ago, the colours got a major overhaul.
Before, they weren't consistently applied -- `default` usually meant the terminal's background colour.

The overhaul lead to `default` strictly meaning **transparent**.
This meant that every colour we applied had to be layered on top of at least the `normal` colour.
The result is more consistent (and prettier :-) colours.

**Comment by jo-so** [comment]
> OK, 6282135c0 should fix the multi-choice prompts.

Yes, it does.

> The "messages" look ok to me.  If I,
> ```
> color message yellow default
> ```

Oh, sorry. You're right. I have `color message green black` in my config.

But I would ask about the changed colour after linewrap, the forth line with
`=mdir/zsh` in the image.

**Comment by flatcap** [comment]
Great.  I'll merge those fixes.
Thanks for your help.

> changed colour after linewrap

Ah.  Good spot.

Now that's a bit trickier -- the Pager is terrifyingly complicated :-)
I'll have a look later.

**Comment by flatcap** [comment]
I've pushed a quick fix for the log colour wrapping: 3baa822e5
It looks OK, but it's not terribly robust.
If you resize the terminal, it gets confused.
That'll have to do for now :-
